### DISCUSSION TITLE
How to persist text spacing and alignment.

### DISCUSSION THREAD

**Discussion by zoutigo** [react-page/react-page#1058] [Q&A]
Hello.

I'm using React-Page inside a form. It is working perfectly from typing datas , then sending to backend , and finally fetching for showing that page(ReadMode) . This is ok.

But when i want to update the content of the page , i'm getting in trouble.
If i add space(not the spacer plugin) between two lines of text, the space is not persisted.

Example:
This is the initial text that i posted with react Page.

`Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.`

When i update, i do the below , adding a space between line . And this is was i see as expected on the form. 
```
`Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard 

dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.`
```
But when i use this updated data to show (ReadOnly) , i'm still getting the first one.
When i open the form again , i see the space was applied. What in form mode is not what i get on ReadOnly.

Any idea how to fix this ?

**Comment by macrozone** [comment]
@zoutigo i assume you use the slate plugin? 

slate will add new paragraphs , when you press enter , or does a \n when you do shift+enter (the later requiring pre-wrap styling).

I suggest you inspect your markup and using the inspector to see why there is a difference. Is the markup different? Are the styles different?

**Reply by zoutigo** [reply]
Thanks for replying.
I've checked with chrome  inspector.
in both cases , i have the same style for the div (the parent)  , and each paragraph

```
p {
    display: block;
    margin-block-start: 1em;
    margin-block-end: 1em;
    margin-inline-start: 0px;
    margin-inline-end: 0px;
}

div {
    position: relative;
    outline: none;
    white-space: pre-wrap;
    overflow-wrap: break-word;
}
```
i still don't see why the empty paragraph (for space) is not rendering as supposed to. 

However , on the edit mode , i see that in the paragraph , inside the deepest span, i have a <br> , witch is not in the readMode

```
  <p data-slate-node="element">
        <span data-slate-node="text">
          <span data-slate-leaf="true">
            <span data-slate-zero-width="n" data-slate-length="0">
              ﻿<br>
            </span>
          </span>
        </span>
      </p>
```

**Comment by zoutigo** [comment]
I found a solution , not the best , with css

```
p:empty: {
    marginTop: '1rem',
  }
```
